nntrainer/tensor/cl_operations/blas_kernels.cpp

```diff
@@ -387,4 +387,268 @@ void sscal_cl(float *X, const unsigned int N, const float alpha) {
 
   } while (false);
 }
+
+void transpose_cl_axis0(const float *in, float *res,
+                        unsigned int input_batch_size,
+                        unsigned int input_channels, unsigned int input_height,
+                        unsigned int input_width) {
+
+  bool result = false;
+
+  do {
+    ClContext::SharedPtrClKernel kernel_transpose_ptr =
+      cl_context_ref.registerClKernel(transpose_cl_kernel_axis0,
+                                      "transpose_cl_axis0");
```

Comment:
There's no need to have three different functions by axis. it can be combined into one.

```suggestion
void transpose_cl_axis(const float *in, float *res,
                        unsigned int input_batch_size,
                        unsigned int input_channels, unsigned int input_height,
                        unsigned int input_width, unsigned int axis) {

  bool result = false;

  do {
    switch (axis) {
    case 0:
      kernel_transpose_ptr = cl_context_ref.registerClKernel(
        transpose_cl_kernel_axis0, "transpose_cl_axis0");
      break;
    case 1:
      kernel_transpose_ptr = cl_context_ref.registerClKernel(
        transpose_cl_kernel_axis0, "transpose_cl_axis1");
      break;
    case 2:
      kernel_transpose_ptr = cl_context_ref.registerClKernel(
        transpose_cl_kernel_axis0, "transpose_cl_axis2");
      break;
    default:
      throw std::invalid_argument("failed to register CL kernel");
      break;
    }
```

Reply:
There would be difference in the work_group_count that we pass for the three axes. Hence I didn't merge them into one.